app/javascript/components/base/base-table-component.vue

```diff
@@ -0,0 +1,61 @@
+<template>
+  <table class="min-w-full divide-y divide-gray-200">
+    <thead class="justify-between">
+      <tr class="bg-yellow-500 border-4 border-yellow-500 py-1">
+        <th v-for="h in table.header">
+          <span class="text-white font-bold">{{ h }}</span>
+        </th>
+        <th v-if="dots" />
+      </tr>
+    </thead>
+    <tbody class="bg-gray-200">
+      <tr
+        v-for="element in table.body"
+        class="bg-white border-4 border-gray-200"
+      >
+        <td
+          v-for="property in table.header"
+          class="content-center py-2"
+        >
+          <p class="content-center text-center ml-2 font-semibold">
+            {{ JSON.parse(element)[property] }}
+          </p>
+        </td>
+        <td
+          v-if="dots"
+          class="content-center"
+        >
+          <dropdown
+            :elements="{
+              edit:true,
+              del:true
+            }"
+            @edit="editIngredient"
+            @del="deleteIngredient"
+          />
+        </td>
+      </tr>
+    </tbody>
+  </table>
+</template>
+
+<script>
+import Dropdown from '../ingredients/base-dropdown-component.vue';
```

Comment:
🚫 **[eslint]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Expected 1 empty line after import statement not followed by another import  import/newline-after-import